homeassistant/components/integration/sensor.py

```diff
@@ -328,6 +347,14 @@ def __init__(
         self._source_entity: str = source_entity
         self._last_valid_state: Decimal | None = None
         self._attr_device_info = device_info
+        self._max_sub_interval: timedelta | None = (
+            max_sub_interval
+            if max_sub_interval is not None and max_sub_interval.total_seconds() > 0
```

Comment:
The check for a positive `total_seconds` should not happen here, it should happen when validating the config schema and in the config flow.

Reply:
This is not a config-validation-like check I guess. This sensor can be initialized with a max_sub_interval of None or a timedelta of 0. If max_sub_interval is not defined or 0 (which can be inserted via DurationSelector in config flow) we want to disable time based integration. I've updated the code to make this hopefully more clear.